src/repo.ts

```diff
@@ -88,23 +175,96 @@ export async function getRepositoryHttpsUrl(
  * of the new release).
  * 3. Switches to that branch.
  *
- * @param projectRepositoryPath - The path to the repository directory.
+ * @param repositoryDirectoryPath - The path to the repository directory.
  * @param args - The arguments.
  * @param args.releaseVersion - The release version.
  */
 export async function captureChangesInReleaseBranch(
-  projectRepositoryPath: string,
+  repositoryDirectoryPath: string,
   { releaseVersion }: { releaseVersion: string },
 ) {
-  await getStdoutFromGitCommandWithin(projectRepositoryPath, [
-    'checkout',
+  await getStdoutFromGitCommandWithin(repositoryDirectoryPath, 'checkout', [
     '-b',
     `release/${releaseVersion}`,
   ]);
-  await getStdoutFromGitCommandWithin(projectRepositoryPath, ['add', '-A']);
-  await getStdoutFromGitCommandWithin(projectRepositoryPath, [
-    'commit',
+  await getStdoutFromGitCommandWithin(repositoryDirectoryPath, 'add', ['-A']);
+  await getStdoutFromGitCommandWithin(repositoryDirectoryPath, 'commit', [
     '-m',
     `Release ${releaseVersion}`,
   ]);
 }
+
+/**
+ * Retrieves the names of the tags in the given repo, sorted by earliest created
+ * to latest created. As this fetches tags from the remote first, you are
+ * advised to only run this once.
+ *
+ * @param repositoryDirectoryPath - The path to the repository directory.
+ * @returns The names of the tags.
+ * @throws If no tags are found and the local git history is incomplete.
+ */
+export async function getTagNames(
+  repositoryDirectoryPath: string,
+): Promise<string[]> {
+  await runGitCommandWithin(repositoryDirectoryPath, 'fetch', ['--tags']);
+
+  // The --merged flag ensures that we only get tags that are parents of or
+  // equal to the current HEAD.
+  const unsortedMergedTagNames = await getLinesFromGitCommandWithin(
+    repositoryDirectoryPath,
+    'tag',
+    ['--merged'],
+  );
+  const sortedComprehensiveTagNames = await getLinesFromGitCommandWithin(
+    repositoryDirectoryPath,
+    'rev-list',
+    ['--tags', '--date-order'],
+  );
```

Comment:
So at first I was going to say that we can get the tags sorted by date with a single command. The `git tag` command accepts a `--sort` parameter that can be used to sort by date.

But, do we want to sort by date here? It seems that the sort order of this list isn't used at all. And even if it was, it would seem more useful to me to sort by semantic order (i.e. `--sort version:refname`) so that backports don't get placed out of order.

Reply:
Yeah, I'm not really using the sorting right now, you're right. I can take the sorting away for now.

Reply:
Or actually, I will keep the sorting for debugging purposes (in case we need to see what it thinks a repo's tags are) but use the sorting order you recommended.

Reply:
If we're not using the order, chronological is fine too, but it would be nice to keep it as one command

Reply:
Updated here: [5c13376](https://github.com/MetaMask/create-release-branch/pull/23/commits/5c13376989bffe01a527af02f51b746743f07f3c)